profile/profile.go

```diff
@@ -140,6 +140,9 @@ type Profile struct { //nolint:maligned // not worth the effort
 	// Lifecycle Management
 	outdated   *abool.AtomicBool
 	lastActive *int64
+
+	// savedInternally is set to true for profiles that are saved internally.
+	savedInternally bool
```

Comment:
I feel like this is just a workaround for fixing an architectural issue with profiles and how they are saved. Otherwise, LGTM, nice we finally found and fixed that